### ISSUE TITLE
ShouldBeEquivalentTo() forces array strict ordering. Should it? I think not.

### ISSUE THREAD

**Issue by dstj** [fluentassertions/fluentassertions.json#25]
If I have an array property, the order of its values needs to be the same when asserting with `Should().BeEquivalentTo()`. I'm arguing that it shouldn't. That would be more in-line with "regular" FluentAssertions.

```
[Test]
public void TestArrayOrder()
{
   var expected =
@"{
   ""data"": [
      { ""value"": 1 },
      { ""value"": 2 }
   ]
}";
   var actual =
@"{
   ""data"": [
      { ""value"": 2 },
      { ""value"": 1 }
   ]
}";

   JsonParsingHelper.Parse(actual).Should().BeEquivalentTo(JsonParsingHelper.Parse(expected));
}
```

results in

```
Message: Expected JSON document {
  "data": [
    {
      "value": 2
    },
    {
      "value": 1
    }
  ]
} to be equivalent to {
  "data": [
    {
      "value": 1
    },
    {
      "value": 2
    }
  ]
}, but the value is different at $.data[0].value.
```

"Regular" FluentAssertions `Should().BeEquivalentTo()` does not enforce strict ordering on array properties. As such, the following test passes:
```
class TestClass
{
    public int[] Values { get; set; }
}

[Test]
public void Test()
{
    var expected = new TestClass { Values = new[] { 1, 2 } };
    var actual = new TestClass { Values = new[] { 2, 1 } };
    actual.Should().BeEquivalentTo(expected);
}
```

This causes me problems because the data I'm comparing comes from a database and the order cannot be guaranteed. I'll need to add ordering in the service I'm testing, but that's modifying the business logic to suit the testing framework. :/

**Comment by dennisdoomen** [comment]
That's an interesting statement. If we would change that, we would also need to expose some options to force strict ordening. But that all depends on how you define _equivalent_.

**Comment by LordLyng** [comment]
Any workarounds?

**Comment by codialsoftware** [comment]
@dennisdoomen why would you like to add any configuration for that? _Equivalence_ is not _be equal to_ so order should not be taken into consideration.

@LordLyng I also encountered this problem and the workaround is actually trivial. Just you `BeEquivalentTo` from `FluentAssertions` itself. It works fine.

**Comment by dennisdoomen** [comment]
> why would you like to add any configuration for that? _Equivalence_ is not _be equal to_ so order should not be taken into consideration.

That's exactly what @LordLyng is proposing...

**Comment by codialsoftware** [comment]
@dennisdoomen Yes, I know. I just wanted to express my surprise for your proposal of exposing additional option. This might make sense for the sake of backward compatibility but how about publishing new major version where default behavior is such as the one we know from `ObjectAssertions.BeEquivalentTo`?

**Comment by TheYarin** [comment]
I agree, this should behave the same as the main library's BeEquivalentTo().

**Comment by mtaghavi2005** [comment]
> @dennisdoomen why would you like to add any configuration for that? _Equivalence_ is not _be equal to_ so order should not be taken into consideration.
> 
> @LordLyng I also encountered this problem and the workaround is actually trivial. Just use `BeEquivalentTo` from `FluentAssertions` itself. It works fine.
> 
> **EDIT**
> This is what I'm currently doing in my code to overcome this issue. Ugly, but works totally fine.
> 
> ```
> ((object) token).Should().BeEquivalentTo(expected)
> ```

Unfortunately, this workaround doesn't work because it will be switched to the main library's BeEquaivalentTo. Like what mentioned [here](https://github.com/fluentassertions/fluentassertions.json#usage), false positives may occur.

**Comment by dennisdoomen** [comment]
If anybody is up for a PR...

**Comment by mattfennerjesi** [comment]
It isn't pretty but this workaround works for me:
`private void AssertJArraysEquivalentIgnoringOrder(JToken actual, JArray expected)
        {
            var newActual = new JArray(actual.Children().OrderBy(c => c.ToString()));
            var newExpected = new JArray(expected.Children().OrderBy(c => c.ToString()));
            JAssert.Should(newActual).BeEquivalentTo(newExpected);
        }`

**Comment by gvdmaaden** [comment]
Hi, has this already been solved in a later release (and if so, what do you need to do)?

Right now I have also a workaround. A bit similar like the one above but mine re-orders all nested arrays in the provided JToken on a recursive way:

```
public static JToken OrderAllNestedArraysInJToken(JToken token)
{
    if (token.Type == JTokenType.Array)
    {
        var array = (JArray)token;               
        token = new JArray(array.Children().OrderBy(a => a.ToString()));                
    }

    if (token.Children().Any())
    {
        for (int i = 0; i < token.Children().Count(); i++)
        {
            var child = token.Children().ElementAt(i);
            child = OrderAllNestedArraysInJToken(child);
            token.Children().ElementAt(i).Replace(child);
        }                
    }
    
    return token;
}
```

**Comment by dennisdoomen** [comment]
No, the JSON version of this library didn't get much attention. In v7 of the main library, we plan to have `System.Text.Json` support.

**Comment by michaelgwelch** [comment]
I'm late to the party but I'm just suggesting that strict ordering is often the correct choice for JSON arrays as they are also used for tuples where order definitely matters. So if the behavior were to change you would definitely want an option to still enforce ordering.
